src/FakeItEasy/Core/ArgumentEqualityComparer.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace FakeItEasy.Core
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    internal class ArgumentEqualityComparer
+    {
+        private readonly IEnumerable<IArgumentEqualityComparer> argumentEqualityComparers;
+
+        public ArgumentEqualityComparer(IEnumerable<IArgumentEqualityComparer> argumentEqualityComparers)
+        {
+            this.argumentEqualityComparers = argumentEqualityComparers;
+        }
+
+        public bool AreEqual(object? expectedValue, object? argumentValue, Type parameterType)
+        {
+            var comparer = this.argumentEqualityComparers
+                .Where(c => c.CanCompare(parameterType))
+                .OrderByDescending(c => c.Priority)
+                .FirstOrDefault();
+
+            if (comparer is null)

```

Comment:
I'd consider adding a default comparer. Minimum priority, always `CanCompare`.

Reply:
So we would use `First()`, assuming there will always be a match?

Reply:
After thinking about it for a while, I'm not sure this would really be better.
The `if (comparer is null)` check is easy to understand, and makes it clear that we handle the case where no custom comparer exists.
If we added a default comparer, we would implicitly fallback to it, which wouldn't be obvious at all from reading the code (well, I guess we could just add a comment, but still... _Explicit is better than implicit_, as Python developers say 😉)